From: Paul Holzinger <git@holzinger.dev>
Date: Tue, 17 Dec 2024 17:20:28 +0100
Subject: test/system: fix "podman play --build private registry" error

Forwarded: https://github.com/containers/podman/pull/24861

When running this test on a system without unqualifiedsearch registries
it will fail with a different error causing the test to fail. to avoid
that case define our own registries.conf that defines quay.io as
registry. This should make the test pass in the debian env.

Signed-off-by: Paul Holzinger <git@holzinger.dev>
---
 test/system/700-play.bats | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/test/system/700-play.bats b/test/system/700-play.bats
index ce5011b..f8d89f6 100644
--- a/test/system/700-play.bats
+++ b/test/system/700-play.bats
@@ -982,6 +982,14 @@ _EOF
     # Remove the local image to make sure it will be pulled again
     run_podman image rm --ignore $from_image
 
+    # The error below assumes unqualified-search registries exist, however the default
+    # distro config may not set some and thus resulting in a different error message.
+    # We could try to match a third or or simply force a know static config to trigger
+    # the right error.
+    local CONTAINERS_REGISTRIES_CONF="$PODMAN_TMPDIR/registries.conf"
+    echo 'unqualified-search-registries = ["quay.io"]' > "$CONTAINERS_REGISTRIES_CONF"
+    export CONTAINERS_REGISTRIES_CONF
+
     _write_test_yaml command=id image=$userimage
     run_podman 125 play kube --build --start=false $TESTYAML
     assert "$output" "=~" \
